tests/main/views/test_brief_response.py

```diff
@@ -550,32 +552,44 @@ def test_can_not_submit_a_brief_response_that_already_been_submitted(self, live_
         data = json.loads(repeat_res.get_data(as_text=True))
         assert data == {'error': 'Brief response must be a draft'}
 
-    def test_can_not_submit_a_brief_response_for_a_framework_that_is_not_live_or_expired(self, live_dos_framework):
-            with self.app.app_context():
-                for framework_status in [status for status in Framework.STATUSES if status not in ('live', 'expired')]:
-
-                    # If a brief response already exists delete the last one. Suppliers can only have one response.
-                    existing_brief_response = db.session.query(BriefResponse).all()
-                    if existing_brief_response:
-                        db.session.delete(existing_brief_response[-1])
-                        db.session.commit()
-
-                    # Make framework live so a brief response can be created.
-                    db.session.execute(
-                        "UPDATE frameworks SET status='live' WHERE slug='digital-outcomes-and-specialists'"
-                    )
-                    self._setup_existing_brief_response()
-
-                    # Set framework status to the invalid status currently under test.
-                    db.session.execute(
-                        "UPDATE frameworks SET status=:status WHERE slug='digital-outcomes-and-specialists'",
-                        {'status': framework_status},
-                    )
-
-                    res = self._submit_brief_response(self.brief_response_id)
-                    data = json.loads(res.get_data(as_text=True))
-                    assert res.status_code == 400
-                    assert data == {'error': 'Brief framework must be live or expired'}
+    @pytest.mark.parametrize('framework_status', ('coming', 'open', 'pending', 'standstill'))
+    def test_can_not_submit_a_brief_response_for_a_framework_that_is_not_live_or_expired(
+            self,
+            live_dos_framework,
+            framework_status
+    ):
+        with self.app.app_context():
+            # If a brief response already exists delete the last one. Suppliers can only have one response.
+            existing_brief_response = db.session.query(BriefResponse).all()
+            if existing_brief_response:
+                db.session.delete(existing_brief_response[-1])
+                db.session.commit()
+
+            # Get old framework status so we can reset it later.
+            dos_framework = db.session.query(Framework).filter_by(slug='digital-outcomes-and-specialists').first()
+            original_status = dos_framework.status
+
+            # Make framework live so a brief response can be created.
+            dos_framework.status = 'live'
```

Comment:
I think you might not need to reset the framework status for each iteration of the test with the code here and on lines 570 and 591-592.
I've dropped a debugger in and the `dos_framework` retrieved on line 569 has a framework status of `live` for each iteration, without having the code to reset it back to its original status.
I'm assuming it's loading the fixture fresh for each iteration of parameters?